File: frontend/types/brief.ts
export type SourceSystem =
  | "fbr_sales"
  | "fbr_mkt"
  | "fbr_video"
  | "fbr_redacao"
  | "fbr_click"
  | "fbr_leads"
  | "fbr_finance"
  | "fbr_dev"
  | "internal_manual";

export interface Brief {
  id: string;
  source_system: SourceSystem;
  source_reference_id?: string | null;
  project_id: string;
  brand_id: string;
  campaign_id?: string | null;
  requester_name: string;
  requester_email?: string | null;
  title: string;
  objective: string;
  audience?: string | null;
  channel?: string | null;
  constraints?: string | null;
  references_summary?: string | null;
  ai_intake_notes?: string | null;
  created_at: string;
  updated_at: string;
}

export interface BriefCreateInput {
  source_system: SourceSystem;
  source_reference_id?: string | null;
  project_id: string;
  brand_id: string;
  campaign_id?: string | null;
  requester_name: string;
  requester_email?: string | null;
  title: string;
  objective: string;
  audience?: string | null;
  channel?: string | null;
  constraints?: string | null;
  references_summary?: string | null;
  ai_intake_notes?: string | null;
}

export interface BriefListResponse {
  data: Brief[];
  meta: {
    count: number;
    limit: number;
    offset: number;
  };
  error: null;
}

export interface BriefCreateResponse {
  data: Brief;
  meta: {
    count: number;
    [key: string]: number | string;
  };
  error: null;
}

File: backend/app/schemas/brief.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, EmailStr

from app.models.enums import SourceSystem
from app.schemas.common import TimestampedSchema


class BriefBase(BaseModel):
    source_system: SourceSystem
    source_reference_id: str | None = None
    project_id: UUID
    brand_id: UUID
    campaign_id: UUID | None = None
    requester_name: str
    requester_email: EmailStr | None = None
    title: str
    objective: str
    audience: str | None = None
    channel: str | None = None
    constraints: str | None = None
    references_summary: str | None = None
    ai_intake_notes: str | None = None


class BriefCreate(BriefBase):
    pass


class BriefRead(BriefBase, TimestampedSchema):
    id: UUID


class BriefListResponse(BaseModel):
    data: list[BriefRead]
    meta: dict[str, int]
    error: None = None


class BriefCreateResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    data: BriefRead
    meta: dict[str, int | str]
    error: None = None
